### PR TITLE
Start transaction with car and tag

### PR DESCRIPTION
**Pull request by mahdibenromdhane**

### CODE REVIEW

File: src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        this.userID = this.formGroup.controls['userID'];
+        this.car = this.formGroup.controls['car'];
+        this.carID = this.formGroup.controls['carID'];
+        this.tag = this.formGroup.controls['tag'];
+        this.tagID = this.formGroup.controls['tagID'];
+        this.user.setValue(Utils.buildUserFullName(this.loggedUser));
+        this.userID.setValue(this.loggedUser.id);
+        this.spinnerCounter = 2;
+        this.loadDefaultCar();
+        this.loadDefaultTag();
```
Comment (LucasBrazi06) [inline src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:94]: You can also have one REST endpoint interface that will return both default car and badge (simplify front and backend).

We should call once the backend per UI displayed (Project Golden Rules # 15 violated :))

image

File: src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        // Register to the answer
+        dialogRef.afterClosed().subscribe((result) => {
+            this.car.setValue(Utils.buildCarName(result[0].objectRef, false));
+            this.carID.setValue(result[0].key);
+        });
+    }
+
+    public startTransaction() {
+        const startTransaction: StartTransaction = {
+            userFullName: this.user.value,
```
Comment (LucasBrazi06) [inline src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:233]: This should be userID right?

File: src/app/types/Transaction.ts [hunk 3]
```diff
@@ -97,6 +99,12 @@ export interface TransactionConsumption {
   cumulatedAmount: number;
 }
 
+export interface StartTransaction {
+userFullName: string;
```
Comment (LucasBrazi06) [inline src/app/types/Transaction.ts:103]: Tabs
userID ?

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
You've been improving in Angular, gut! ;-)
My comments.

File: src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts (hunk 2 above)
Comment (mahdibenromdhane) [reply src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:233]: We only need the userFullName to display in the confirmation pop up. if we pass the userID we will need to call the server to get the Object.

**Comment by mahdibenromdhane** [comment]
> You've been improving in Angular, gut! ;-)
> My comments.

Thanks :D

File: src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts (hunk 2 above)
Comment (LucasBrazi06) [reply src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:233]: But why is there a carID and tagID below in that case?

Comment (mahdibenromdhane) [reply src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:233]: We pass the carID, TagID and userFullName to the method startTransactionForUser in the table-charging-stations-start-transaction-action. 
- CarID and TagID go to the server to start transaction 
- userFullName to display in the confirmation Modal

Comment (LucasBrazi06) [reply src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:233]: Ok got it, but when the admin chooses another user, you should also pass the user ID right?

Comment (mahdibenromdhane) [reply src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:233]: No, only the TagID go to the server in all cases.

Comment (mahdibenromdhane) [reply src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts:233]: We always retrieve the user from tag ID.

File: src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts (hunk 1 above)
Comment (mahdibenromdhane) [reply src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts]: Done :)

**Comment by LucasBrazi06** [comment]
Conflicts to resolve due to the latest merge.

**Comment by mahdibenromdhane** [comment]
> Conflicts to resolve due to the latest merge.

Resolved.

**Review by LucasBrazi06 — APPROVED** [review]
Feedback will be sent by email

**Pull request merged**